be/src/vec/core/block.cpp

```diff
@@ -321,13 +322,16 @@ void Block::check_number_of_rows(bool allow_null_columns) const {
         if (rows == -1) {
             rows = size;
         } else if (rows != size) {
-            LOG(FATAL) << fmt::format("Sizes of columns doesn't match: {}:{},{}:{}",
-                                      data.front().name, rows, elem.name, size);
+            LOG(FATAL) << fmt::format("Sizes of columns doesn't match: {}:{},{}:{}, col size: {}",
+                                      data.front().name, rows, elem.name, size, each_col_size());
         }
     }
 }
 
 size_t Block::rows() const {
+    if (_effective_col != INT_MIN) {
```

Comment:
use column const during filter block. and then replace column after filter block.
Then do not need add this effective col any more.

Reply:
done